src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -67,20 +73,120 @@ public AuthenticatedEncryptionProvider(SecurityKey key, string algorithm)
             if (string.IsNullOrWhiteSpace(algorithm))
                 throw LogHelper.LogArgumentNullException(nameof(algorithm));
 
-            _authenticatedkeys = new Lazy<AuthenticatedKeys>(CreateAuthenticatedKeys);
-            _hmacAlgorithm = GetHmacAlgorithm(algorithm);
             Key = key;
             Algorithm = algorithm;
             _cryptoProviderFactory = key.CryptoProviderFactory;
 
+            if (SupportedAlgorithms.IsSupportedEncryptionAlgorithm(algorithm, key))
+            {
+                if (SupportedAlgorithms.IsAesGcm(algorithm))
+                {
+#if NETSTANDARD2_0
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    throw LogHelper.LogExceptionMessage(new PlatformNotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10713, algorithm)));
+#endif
+                    InitializeUsingAesGcm();
+                }
+                else
+                    InitializeUsingAesCbc();
+            }
+            else
+                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10668, GetType(), algorithm, key)));
+        }
+
+        private void InitializeUsingAesGcm()
+        {
+            ValidateKeySize(Key, Algorithm);
```

Comment:
ValidateKeySize is a protected virtual method and shouldn't be called from a ctor. See https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1549

Reply:
Fixed it [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1606/commits/b60b69315e3b1db70e014df09b507c0ce7c3d95c)